lib/matplotlib/pyplot.py

```diff
@@ -543,9 +531,26 @@ def make_active(event):
         _pylab_helpers.Gcf.set_active(figManager)
         figManager.canvas.figure.number = num
 
+        if _INSTALL_FIG_OBSERVER:
+            figManager.canvas.figure.add_callback(_auto_draw_if_interactive)
```

Comment:
Because this is a possibly annoying feature for people embedding mpl into larger applications.  They should be able control on what rep-cycle they trigger re-draws.  They now have an easy way to tell if they need to, but we should not impose on them re-draws they do not want.  Ex setting a timed call back to every .05s check if anything is stale and if so trigger a re-draw of just that figure.